File: chat-uikit/src/main/java/io/agora/chat/uikit/chatthread/EaseChatThreadRole.java
package io.agora.chat.uikit.chatthread;

public enum EaseChatThreadRole {
    /**
     * thread member
     */
    MEMBER,
    /**
     * thread creator
     */
    CREATOR,
    /**
     * group admin
     */
    GROUP_ADMIN,
    /**
     * Unknown identity
     */
    UNKNOWN;

    public static EaseChatThreadRole getThreadRole(int role) {
        EaseChatThreadRole threadRole = UNKNOWN;
        switch (role) {
            case 0 :
                threadRole = MEMBER;
                break;
            case 1 :
                threadRole = CREATOR;
                break;
            case 2 :
                threadRole = GROUP_ADMIN;
                break;
        }
        return threadRole;
    }
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/feature/thread/widgets/EaseChatThreadRole.kt
package io.agora.uikit.feature.thread.widgets

enum class EaseChatThreadRole{
    /**
     * thread member
     */
    MEMBER,

    /**
     * thread creator
     */
    CREATOR,

    /**
     * group admin
     */
    GROUP_ADMIN,

    /**
     * Unknown identity
     */
    UNKNOWN;

    companion object {
        fun getThreadRole(role: Int): EaseChatThreadRole {
            var threadRole = UNKNOWN
            when (role) {
                0 -> threadRole = MEMBER
                1 -> threadRole = CREATOR
                2 -> threadRole = GROUP_ADMIN
            }
            return threadRole
        }
    }
}
